deploy/health_check/main.py

```diff
@@ -0,0 +1,83 @@
+"""
+Health check for Horizon.
+
+Healthy: Corresponding stellar-core is 'Synced'
+Unhealthy: Failed/Total requests is > 0.1
+"""
+import json
+import time
+
+import requests
+from flask import Flask
+from flask_cors import CORS
+
+import config
+
+
+app = Flask(__name__)
+CORS(app)
+start_timestamp = time.time()
+
+# For 'health', true is healthy
+health = True
+
+
+def check_core():
+    # Check if the stellar core is synced
+    global health
+    try:
+        r = requests.get(config.CORE_INFO_URL)
+        r.raise_for_status()
+
+        health = (r.json()['info']['state'] == 'Synced!')
+        if health:
+            return make_reply('Core is synced', 200)
+        return make_reply('Core is not synced', 503)
+    except Exception as e:
```

Comment:
it's bad practice to do a catch-all `Exception` block. catch specific exceptions (see requests docs). since this function will run as an http request handler it is ok to crash

Reply:
Letting it crash will return 500 and no info, we wanted to return 503 + description of the error, timestamp, and build